libraries/AP_CANManager/AP_CANManager.cpp

```diff
@@ -318,15 +320,13 @@ void AP_CANManager::log_text(AP_CANManager::LogLevel loglevel, const char *tag,
 }
 
 // log retrieve method used by file sys method to report can log
-uint32_t AP_CANManager::log_retrieve(char* data, uint32_t max_size) const
+void AP_CANManager::log_retrieve(ExpandingString *str) const
```

Comment:
should this be ExpandingString& ?

Reply:
it's a pointer so we don't need the header in all the interfaces. I'm trying to avoid having more headers that end up getting included in nearly all our code, which slows down the build

Reply:
yep, fixed